ts/indicators/aroon.src.ts

```diff
@@ -0,0 +1,230 @@
+        },
+        dataGrouping: {
+            approximation: 'averages'
+        }
+    },
+    /**
+     * @lends Highcharts.Series#
+     */
+    H.merge(multipleLinesMixin, {
+        nameBase: 'Aroon',
+        pointArrayMap: ['y', 'aroonDown'],
+        pointValKey: 'y',
+        linesApiNames: ['aroonDown'],
+        getValues: function (
+            series: Highcharts.AroonIndicator,
+            params: Highcharts.AroonIndicatorParamsOptions
+        ): Highcharts.IndicatorMultipleValuesObject {
+            var period: number = params.period,
+                xVal: (Array<number>|undefined) = series.xData,
+                yVal: Array<Array<number>> = series.yData,
+                yValLen = yVal ? yVal.length : 0,
+                // 0- date, 1- Aroon Up, 2- Aroon Down
+                AR: Array<Array<number>> = [],
+                xData: Array<number> = [],
+                yData: Array<Array<number>> = [],
+                slicedY: Array<Array<number>>,
+                low = 2,
+                high = 1,
+                aroonUp: number,
+                aroonDown: number,
+                xLow: number,
+                xHigh: number,
+                i: number;
+
+            // For a N-period, we start from N-1 point, to calculate Nth point
+            // That is why we later need to comprehend slice() elements list
+            // with (+1)
+            for (i = period - 1; i < yValLen; i++) {
+                slicedY = yVal.slice(i - period + 1, i + 2);
+
+                xLow = getExtremeIndexInArray(slicedY.map(
+                    function (elem): number {
+                        return H.pick(elem[low], (elem as any));
+                    }), 'min');
+
+                xHigh = getExtremeIndexInArray(slicedY.map(
+                    function (elem): number {
+                        return H.pick(elem[high], (elem as any));
+                    }), 'max');
+
+                aroonUp = (xHigh / period) * 100;
+                aroonDown = (xLow / period) * 100;
+
+                if ((xVal as any)[i + 1]) {
+                    AR.push([(xVal as any)[i + 1], aroonUp, aroonDown]);
+                    xData.push((xVal as any)[i + 1]);
+                    yData.push([aroonUp, aroonDown]);
+                }
+            }
+
+            return {
+                values: AR,
+                xData: xData,
+                yData: yData
+            };
+        }
+    })
+);
+
+/**
+ * A Aroon indicator. If the [type](#series.aroon.type) option is not
+ * specified, it is inherited from [chart.type](#chart.type).
+ *
+ * @extends   series,plotOptions.aroon
+ * @since     7.0.0
+ * @product   highstock
+ * @excluding allAreas, colorAxis, compare, compareBase, dataParser, dataURL,
+ *            joinBy, keys, navigatorOptions, pointInterval, pointIntervalUnit,
+ *            pointPlacement, pointRange, pointStart, showInNavigator, stacking
+ * @apioption series.aroon
+ */
+
+''; // to include the above in the jsdocs
```

Comment:
Comment not quite right. TypeScript would remove the doclet from the JS output, if no code line is following.